docs/book/middleware.md

```diff
@@ -0,0 +1,53 @@
+# PSR-7 Middleware
+For every bus system a middleware exists and one Middleware to rule them all. If you use JSON or XML in the request body
+for your message data, you have to convert this data to an array before you can call the middleware.
+
+> Note: The middleware uses an array for the message data
+
+## CommandMiddleware
+The `CommandMiddleware` dispatches the message data to the command bus system. This middleware needs an request attribute 
+(`$request->getAttribute(\Prooph\Psr7Middleware\CommandMiddleware::NAME_ATTRIBUTE)`) called `prooph_command_name`. 
+This name is used for the `\Prooph\Common\Messaging\MessageFactory` to create the `\Prooph\Common\Messaging\Message` 
+object. The data for the command is extracted from the body of the request (`$request->getParsedBody()`) and must be an 
+array.
+
+## QueryMiddleware
+The `QueryMiddleware` dispatches the message data to the query bus system. This middleware needs an request attribute 
+(`$request->getAttribute(\Prooph\Psr7Middleware\QueryMiddleware::NAME_ATTRIBUTE)`) called `prooph_query_name`. 
+This name is used for the `\Prooph\Common\Messaging\MessageFactory` to create the `\Prooph\Common\Messaging\Message` 
+object. The data for the query is extracted from the body of the request (`$request->getParsedBody()`) and must be an 
```

Comment:
My Query-API uses GET to get the query name, filter parameters and pagination parameters and POST for the payload (additional information, needed for the query). This has some reasons:

1) It should be generic for my API, I don't want to have special controllers for every query
2) The filter parameters contains f.e. 'sort' this is used to sort the query result. On the other hand 'sort' can be a required payload parameter. So when I combine GET & POST I can have two different sort values with two different meanings. First (GET) is for the sorting result, second (POST) can be a required parameter for the query object (payload).